lib/cequel/metal/keyspace.rb

```diff
@@ -311,16 +311,19 @@ def extract_hosts_and_port(configuration)
             warn "Specifying a hostname as host:port is deprecated. Specify " \
                  "only the host IP or hostname in :hosts, and specify a " \
                  "port for all nodes using the :port option."
-            ports << port.to_i
+            ports << port
           end
         end
 
         if ports.size > 1
           fail ArgumentError, "All Cassandra nodes must listen on the same " \
                "port; specified multiple ports #{ports.join(', ')}"
+        elsif ports.size != 1
+          fail ArgumentError, "Port for cassandra nodes not specified"
         end
+        
 
-        [hosts, ports.first || 9042]
+        [hosts, Integer(ports.first) || 9042]
```

Comment:
You should probably use `Integer(ports.first || 9042)` instead because integer will fail to parse nil if no ports are supplied